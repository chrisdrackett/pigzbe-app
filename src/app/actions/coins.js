import {COINS} from '../constants';
import {apiURL} from '../selectors';

export const EXCHANGE_LOAD = 'EXCHANGE_LOAD';

export const getExchange = () => async (dispatch, getState) => {
    try {
        const api = apiURL(getState());
        const values = await (await fetch(`${api}/compare?coins=${COINS.toString()}`)).json();

        dispatch({type: EXCHANGE_LOAD, payload: {
            exchange: values,
            error: null
        }});

    } catch (error) {
        dispatch({type: EXCHANGE_LOAD, payload: {
            exchange: null,
            error: new Error('Network error')
        }});
    }
};
