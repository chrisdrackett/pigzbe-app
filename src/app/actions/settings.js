import {apiURL} from '../selectors';
import Storage from '../utils/storage';
import {STORAGE_KEY_SETTINGS} from '../constants';

export const SETTINGS_UPDATE = 'SETTINGS_UPDATE';
export const SETTINGS_CLEAR = 'SETTINGS_CLEAR';

export const loadSettings = () => async dispatch => {
    console.log('1b. loadSettings');
    try {
        const data = await Storage.load(STORAGE_KEY_SETTINGS);
        console.log(JSON.stringify(data, null, 2));
        dispatch({type: SETTINGS_UPDATE, ...data});
    } catch (error) {
        console.log(error);
    }
};

export const settingsSave = () => async (dispatch, getState) => {
    try {
        const data = getState().settings;
        await Storage.save(STORAGE_KEY_SETTINGS, data);
    } catch (error) {
        console.log(error);
    }
};

export const settingsUpdate = data => async (dispatch) => {
    try {
        dispatch({type: SETTINGS_UPDATE, ...data});
        await dispatch(settingsSave());
    } catch (error) {
        console.log(error);
    }
};

export const settingsFirstTime = () => dispatch => {
    dispatch(settingsUpdate({firstTime: false}));
};

export const settingsEnableTouchId = value => dispatch => {
    dispatch(settingsUpdate({enableTouchId: value}));
};

export const settingsToggleSubscribe = subscribe => async (dispatch, getState) => {
    try {
        dispatch(settingsUpdate({subscribe}));
        const api = apiURL(getState());
        const {email} = getState().settings;
        await (await fetch(`${api}/email/${subscribe ? 'subscribe' : 'unsubscribe'}`, {
            method: 'POST',
            body: JSON.stringify({email})
        })).json();
    } catch (e) {
        console.log(e);
        dispatch(settingsUpdate({subscribe: !subscribe}));
    }
};

export const settingsClear = () => () => {
    return Storage.clear(STORAGE_KEY_SETTINGS);
};
