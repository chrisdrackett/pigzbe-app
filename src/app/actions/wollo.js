import {
    loadAccount,
    setServer,
    isValidPublicKey,
    paymentHistory,
    paymentInfo,
    sendPayment,
    getBalance,
    getMinBalance,
    checkHasGas,
    checkAssetTrusted,
    trustAsset,
    createAccount,
    TransactionBuilder,
    trustAssetTransaction,
    submitTransaction,
    Keypair
} from '@pigzbe/stellar-utils';
import {
    strings,
    ASSET_CODE
} from '../constants';
import Keychain from '../utils/keychain';
import {wolloAsset} from '../selectors';
import {createKeypair, appError, updateKidBalance} from './';

export const WOLLO_LOADING = 'WOLLO_LOADING';
export const WOLLO_ERROR = 'WOLLO_ERROR';
export const WOLLO_UPDATE_ACCOUNT = 'WOLLO_UPDATE_ACCOUNT';
export const WOLLO_UPDATE_BALANCE = 'WOLLO_UPDATE_BALANCE';
export const WOLLO_UPDATE_XLM = 'WOLLO_UPDATE_XLM';
export const WOLLO_UPDATE_PAYMENTS = 'WOLLO_UPDATE_PAYMENTS';
export const WOLLO_SENDING = 'WOLLO_SENDING';
export const WOLLO_SEND_COMPLETE = 'WOLLO_SEND_COMPLETE';
export const WOLLO_SEND_STATUS = 'WOLLO_SEND_STATUS';
export const WOLLO_SEND_RESET = 'WOLLO_SEND_RESET';
export const WOLLO_KEYPAIR = 'WOLLO_KEYPAIR';
export const WOLLO_KEYPAIR_SAVED = 'WOLLO_KEYPAIR_SAVED';

export const getWolloBalance = account => getBalance(account, ASSET_CODE);

export const wolloLoading = loading => ({type: WOLLO_LOADING, loading});
export const wolloSending = sending => ({type: WOLLO_SENDING, sending});
export const wolloSendComplete = sendComplete => ({type: WOLLO_SEND_COMPLETE, sendComplete});
export const wolloSendStatus = sendStatus => ({type: WOLLO_SEND_STATUS, sendStatus});
export const wolloSendReset = () => ({type: WOLLO_SEND_RESET});

export const wolloError = error => ({type: WOLLO_ERROR, error});

export const setHorizonURI = uri => () => {
    console.log('setHorizonURI:', uri);
    setServer(uri);
};

const updateBalance = balance => ({type: WOLLO_UPDATE_BALANCE, balance});

const updateXLM = account => dispatch => {
    const balanceXLM = getBalance(account);
    const minXLM = getMinBalance(account, 10000);
    console.log('minXLM', minXLM);
    const hasGas = checkHasGas(account);
    dispatch({type: WOLLO_UPDATE_XLM, balanceXLM, minXLM, hasGas});
};

export const loadWallet = publicKey => async (dispatch, getState) => {
    console.log('loadWallet');
    const key = publicKey || getState().keys.publicKey;
    try {
        if (key) {
            const account = await loadAccount(key);
            console.log('account', account);
            dispatch({type: WOLLO_UPDATE_ACCOUNT, account});
            dispatch(updateBalance(getWolloBalance(account)));

            const asset = wolloAsset(getState());
            const isTrusted = checkAssetTrusted(account, asset);
            if (!isTrusted) {
                await trustAsset(getState().keys.secretKey, asset);
            }
            dispatch(updateXLM(account));
        }
    } catch (error) {
        console.log('Could not load wallet with publicKey', key);
    }
};

export const refreshBalance = kidAddress => async (dispatch, getState) => {
    const key = getState().keys.publicKey;
    try {
        if (key) {
            const account = await loadAccount(key);
            dispatch({type: WOLLO_UPDATE_ACCOUNT, account});
            dispatch(updateBalance(getWolloBalance(account)));
            dispatch(updateXLM(account));
        }
        if (kidAddress) {
            const account = await loadAccount(kidAddress);
            dispatch(updateKidBalance(kidAddress, getWolloBalance(account)));
        }
    } catch (error) {
        console.log('Could not load wallet with publicKey', key);
    }
};

export const loadPayments = address => async (dispatch, getState) => {
    const {publicKey} = getState().keys;
    const key = address || publicKey;

    dispatch(wolloLoading(true));
    dispatch(wolloError(null));

    try {
        const rawData = await paymentHistory(key);
        const filteredData = rawData.filter(p => p.type !== 'account_merge');
        const payments = await Promise.all(filteredData.map(p => paymentInfo(p, key)));
        dispatch({type: WOLLO_UPDATE_PAYMENTS, payments});
    } catch (error) {
        console.log(error);
    }

    dispatch(wolloLoading(false));
};


export const sendWollo = (destination, amount, memo) => async (dispatch, getState) => {
    console.log('sendWollo', destination, amount, memo);
    if (!isValidPublicKey(destination)) {
        dispatch(wolloError(strings.transferErrorInvalidKey));
        dispatch(appError(strings.transferErrorInvalidKey));
        return;
    }

    if (!amount || isNaN(amount) || Number(amount) === 0) {
        dispatch(wolloError(strings.transferErrorInvalidAmount));
        dispatch(appError(strings.transferErrorInvalidAmount));
        return;
    }

    dispatch(wolloError(null));
    dispatch(wolloSendComplete(false));
    dispatch(wolloSending(true));
    dispatch(wolloSendStatus(strings.transferStatusChecking));

    const {secretKey, publicKey} = getState().keys;

    let destAccount;

    try {
        destAccount = await loadAccount(destination);
    } catch (e) {
        console.log(e);
    }

    if (!destAccount) {
        dispatch(wolloSending(false));
        dispatch(wolloSendStatus(null));
        dispatch(wolloError(strings.transferStatusInvalidDestination));
        dispatch(appError(strings.transferStatusInvalidDestination));
        return;
    }

    const asset = wolloAsset(getState());

    const isTrusted = checkAssetTrusted(destAccount, asset);

    if (!isTrusted) {
        dispatch(wolloSending(false));
        dispatch(wolloSendStatus(null));
        dispatch(wolloError(strings.transferErrorTrust));
        dispatch(appError(strings.transferErrorTrust));
        return;
    }

    dispatch(wolloSendStatus(strings.transferStatusSending));

    let result;

    try {
        result = await sendPayment(secretKey, destination, amount, memo, asset);
    } catch (e) {
        console.error(e);
    }

    if (!result) {
        dispatch(wolloSending(false));
        dispatch(wolloSendStatus(null));
        dispatch(wolloError(strings.transferStatusFailed));
        dispatch(appError(strings.transferStatusFailed));
        return;
    }

    dispatch(refreshBalance());
    dispatch(wolloSending(false));
    dispatch(wolloSendComplete(true));
    dispatch(wolloSendStatus(strings.transferStatusComplete));
    dispatch(wolloError(null));
    dispatch(loadWallet(publicKey));
};

export const fundKidAccount = (memo, address, startingBalance) => async (dispatch, getState) => {
    console.log('fundKidAccount', memo, address, startingBalance);
    // const {publicKey, secretKey} = getState().keys;
    const {secretKey} = getState().keys;
    const kidSecretKey = await Keychain.load(`secret_${address}`);
    try {
        const keypair = Keypair.fromSecret(kidSecretKey);
        const destination = keypair.publicKey();

        const account = await createAccount(secretKey, destination, startingBalance, memo);
        console.log('account', account);

        const txb = new TransactionBuilder(account);

        const asset = wolloAsset(getState());
        trustAssetTransaction(txb, asset);

        const transaction = txb.build();
        transaction.sign(keypair);
        const result = await submitTransaction(transaction);
        console.log('result', result);
    } catch (error) {
        console.log(error);
    }
};

export const createKidAccount = (memo, nickname, startingBalance) => async dispatch => {
    console.log('createKidAccount', nickname);

    try {
        const keypair = await dispatch(createKeypair());
        const destination = keypair.publicKey();
        console.log('createKidAccount destination', destination);

        await Keychain.save(`secret_${destination}`, keypair.secret());

        console.log('createKidAccount startingBalance', startingBalance);

        await dispatch(fundKidAccount(`${memo}${nickname}`, destination, startingBalance));

        return destination;

    } catch (error) {
        console.log(error);
    }

    return null;
};
