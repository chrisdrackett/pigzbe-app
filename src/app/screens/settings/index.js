import React, {Component} from 'react';
import {connect} from 'react-redux';
import {View} from 'react-native';
import {
    loadEscrow,
    // settingsClear,
    // settingsUpdate,
    settingsEnableTouchId,
    settingsToggleSubscribe,
    authLogout
} from '../../actions';
import Button from '../../components/button';
import {
    strings,
    SCREEN_BALANCE,
    SCREEN_CLAIM,
    SCREEN_ESCROW,
    PRIVACY_URL,
    SCREEN_CHANGE_PASSCODE
} from '../../constants';
import openURL from '../../utils/open-url';
import StepModule from '../../components/step-module';
// import Paragraph from '../../components/paragraph';
import Checkbox from '../../components/checkbox';

export class Settings extends Component {
    componentWillMount() {
        this.props.dispatch(loadEscrow());
    }

    onBack = () => this.props.navigation.navigate(SCREEN_BALANCE)

    onClaim = () => this.props.navigation.navigate(SCREEN_CLAIM)

    onEscrow = () => this.props.navigation.navigate(SCREEN_ESCROW)

    onSubscribe = () => this.props.dispatch(settingsToggleSubscribe(!this.props.subscribe));

    onEnableTouchId = () => this.props.dispatch(settingsEnableTouchId(!this.props.enableTouchId));

    onLogout = () => {
        this.props.dispatch(authLogout());
        // this.props.dispatch(settingsClear());
    }

    onPrivacy = () => openURL(PRIVACY_URL)

    onChangePasscode = () => this.props.navigation.navigate(SCREEN_CHANGE_PASSCODE)

    render() {
        const {
            touchIdSupport,
            enableTouchId,
            subscribe,
            // email,
            // phone,
            // country,
            escrow,
        } = this.props;

        return (
            <StepModule
                title={'Settings'}
                icon="settings"
                onBack={this.onBack}
                pad
                paddingTop={20}
            >
                <View>
                    {escrow && (
                        <Button
                            label={'View Escrow'}
                            onPress={this.onEscrow}
                        />
                    )}
                    <Button
                        label={'Claim Your Wollo'}
                        onPress={this.onClaim}
                        style={{marginBottom: 30}}
                    />
                    {touchIdSupport && (
                        <Checkbox
                            alt
                            text={touchIdSupport === 'FaceID' ? 'Enable Face ID' : 'Enable Touch ID'}
                            value={enableTouchId}
                            onValueChange={this.onEnableTouchId}
                        />
                    )}
                    <Checkbox
                        alt
                        text={strings.accountMailingListOptIn}
                        value={subscribe}
                        onValueChange={this.onSubscribe}
                    />
                    {/* <Paragraph>{'touchIdSupport: '}{touchIdSupport}</Paragraph> */}
                    {/* <Paragraph>{'enableTouchId: '}{enableTouchId ? 'Yes' : 'No'}</Paragraph> */}
                    {/* <Paragraph>{'subscribe: '}{subscribe ? 'Yes' : 'No'}</Paragraph> */}
                    {/* <Paragraph>{'email: '}{email}</Paragraph> */}
                    {/* <Paragraph>{'phone: '}+{country}{phone}</Paragraph> */}
                    <Button
                        theme="outline"
                        label={'Change passcode'}
                        onPress={this.onChangePasscode}
                        style={{marginTop: 20}}
                    />
                    <Button
                        theme="outline"
                        label={strings.accountLogoutButtonLabel}
                        onPress={this.onLogout}
                        // style={{marginTop: 30}}
                    />
                    {/* <Button
                        theme="plain"
                        label={strings.accountPrivacyButtonLabel}
                        onPress={this.onPrivacy}
                    /> */}
                </View>
            </StepModule>
        );
    }
}

export default connect(
    state => ({
        touchIdSupport: state.auth.touchIdSupport,
        enableTouchId: state.settings.enableTouchId,
        subscribe: state.settings.subscribe,
        email: state.settings.email,
        phone: state.settings.phone,
        country: state.settings.country,
        escrow: state.escrow.escrowPublicKey,
    })
)(Settings);
