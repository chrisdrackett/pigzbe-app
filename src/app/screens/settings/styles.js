import {StyleSheet} from 'react-native';
import {
    input,
    color,
    fontFamily
} from '../../styles';

export default StyleSheet.create({
    outer: {
        backgroundColor: color.blue,
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
    },
    avatar: {
        alignItems: 'center',
        justifyContent: 'center',
        marginBottom: 20
    },
    avatarText: {
        fontFamily,
        color: color.whiteOpacity60,
        fontSize: 14,
        textAlign: 'center',
        marginTop: 20
    },
    loader: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'rgba(255, 255, 255, 0.8)',
        position: 'absolute',
        left: 0,
        top: 0,
        right: 0,
        bottom: 0
    },
    title: {
        fontFamily,
        color: color.white,
        fontSize: 16,
        fontWeight: 'bold',
        textAlign: 'center',
        marginBottom: 0,
        marginTop: 10
    },
    error: {
        fontFamily,
        color: color.red,
        fontSize: 18
    },
    input,
    inputError: {
        ...input,
        borderColor: color.red
    }
});
