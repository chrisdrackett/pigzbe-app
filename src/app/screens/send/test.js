import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Send from './';
import wollo from '../../reducers/wollo';
import coins from '../../reducers/coins';
import settings from '../../reducers/settings';
import {mockStore} from '../../../setupTests';

const props = {
};

describe('Send', () => {
    test('renders correctly', () => {
        renderer.create(
            <Provider store={mockStore({wollo, coins, settings})}>
                <Send {...props}/>
            </Provider>
        );
    });
});
