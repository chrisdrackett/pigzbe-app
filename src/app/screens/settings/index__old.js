import React, {Component} from 'react';
import {connect} from 'react-redux';
import {
    Text,
    View,
    TouchableOpacity,
    ScrollView
} from 'react-native';
import {
    profileUpdate,
    profileAvailable,
    profileClear,
    authLogout
} from '../../actions';
import styles from './styles';
import Button from '../../components/button';
import TextInput from '../../components/text-input';
import Loader from '../../components/loader';
import Avatar from '../../components/avatar';
import Checkbox from '../../components/checkbox';
import Header from '../../components/header';
import Container from '../../components/container';
import KeyboardAvoid from '../../components/keyboard-avoid';
import {pickImage} from '../../utils/image-picker';
import isEmail from '../../utils/is-email';
import {
    strings,
    SCREEN_BALANCE,
    PRIVACY_URL
} from '../../constants';
import openURL from '../../utils/open-url';

class Settings extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name: props.name,
            email: props.email,
            image: props.image,
            subscribe: props.subscribe,
            validName: true,
            validEmail: true,
            isUpdating: false
        };
    }

    validate() {
        const {
            name,
            email
        } = this.state;

        const validName = name && !!name.trim();
        const validEmail = email && isEmail(email);

        this.setState({
            validName,
            validEmail
        });

        return validName && validEmail;
    }

    async save() {
        const isValid = this.validate();

        if (!isValid) {
            return;
        }

        const {dispatch, navigation} = this.props;

        const {
            name,
            email,
            image,
            subscribe
        } = this.state;

        this.setState({isUpdating: true});

        await dispatch(profileUpdate({name, email, image, subscribe}));

        this.setState({isUpdating: false}, () => {
            dispatch(profileAvailable(true));
            if (navigation) {
                navigation.navigate(SCREEN_BALANCE);
            }
        });
    }

    onPressAvatar = async () => {
        try {
            const {uri} = await pickImage();
            this.setState({image: uri});
        } catch (e) {
            console.log('error', e);
        }
    }

    render() {
        const {
            dispatch,
            hasProfile,
            error,
            navigation
        } = this.props;

        const {
            email,
            subscribe,
            validEmail,
            isUpdating
        } = this.state;

        return (
            <View style={styles.outer}>
                <ScrollView bounces={false} style={{flex: 1}} contentContainerStyle={{flexGrow: 1}}>
                    <Container body style={{
                        paddingBottom: hasProfile ? 10 : 20,
                    }}>
                        <KeyboardAvoid>
                            <View>
                                <Header/>
                                <Text style={styles.title}>{'Settings'}</Text>
                                {/* <TouchableOpacity
                                    style={styles.avatar}
                                    onPress={this.onPressAvatar}>
                                    <Avatar select image={image}/>
                                    <Text style={styles.avatarText}>
                                        {hasProfile ? strings.accountChangeImage : strings.accountAddImage}
                                    </Text>
                                </TouchableOpacity> */}
                                {/* <TextInput
                                    error={!validName}
                                    placeholder={strings.accountNamePlaceholder}
                                    value={name}
                                    onChangeText={value => this.setState({name: value})}
                                /> */}
                                {/* <TextInput
                                    keyboardType="email-address"
                                    error={!validEmail}
                                    placeholder={strings.accountEmailPlaceholder}
                                    value={email}
                                    onChangeText={value => this.setState({email: value})}
                                /> */}
                                {/* <Checkbox
                                    text={strings.accountMailingListOptIn}
                                    value={subscribe}
                                    onValueChange={() => this.setState({subscribe: !this.state.subscribe})}
                                /> */}
                            </View>
                        </KeyboardAvoid>
                        <View>
                            <Button
                                label={hasProfile ? strings.accountSaveButtonLabel : strings.accountSubmitButtonLabel}
                                onPress={() => this.save()}
                            />
                            {hasProfile ? (
                                <Button
                                    label={strings.accountCancelButtonLabel}
                                    onPress={() => navigation.navigate(SCREEN_BALANCE)}
                                />
                            ) : null}
                            {hasProfile ? (
                                <Button
                                    label={strings.accountLogoutButtonLabel}
                                    plain
                                    onPress={() => {
                                        dispatch(authLogout());
                                        dispatch(profileClear());
                                    }}
                                />
                            ) : (
                                <Button
                                    label={strings.accountPrivacyButtonLabel}
                                    plain
                                    onPress={() => openURL(PRIVACY_URL)}
                                />
                            )}
                            {error && (
                                <Text style={styles.error}>{error.message}</Text>
                            )}
                        </View>
                        <Loader
                            isLoading={isUpdating}
                        />
                    </Container>
                </ScrollView>
            </View>
        );
    }
}

export default connect(
    state => ({
        enableTouchId: state.settings.enableTouchId,
        email: state.settings.email,
        phone: state.settings.phone,
        country: state.settings.country,
        subscribe: state.settings.subscribe,
    })
)(Settings);
