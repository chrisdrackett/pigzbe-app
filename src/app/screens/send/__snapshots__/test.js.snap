// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgb(0, 50, 120)",
        "flex": 1,
        "justifyContent": "space-between",
      },
      Object {
        "width": 750,
      },
      Object {
        "display": "flex",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(21, 74, 150)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 242,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(0, 50, 120)",
        "justifyContent": "center",
        "minHeight": 30,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/app/components/logo/images/pigzbe_logo_inapp.png",
        }
      }
      style={
        Object {
          "height": 25,
          "marginBottom": 10,
          "marginTop": 34,
          "width": 65,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgb(255, 255, 255)",
            "fontFamily": "Poppins",
            "fontSize": 22,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Transfer Wollo
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/app/components/step-header/images/transfer.png",
        }
      }
      style={
        Object {
          "height": 100,
          "marginBottom": -30,
          "marginTop": 8,
          "width": 200,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexGrow": 1,
          },
          Object {
            "bottom": 0,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgb(0, 50, 120)",
              "flex": 1,
              "justifyContent": "space-between",
            },
            Object {
              "width": 750,
            },
            Array [
              Object {
                "alignSelf": "center",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "width": "88.75%",
              },
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
              },
            ],
          ]
        }
      >
        <RCTScrollView
          bounces={false}
          contentContainerStyle={
            Object {
              "flexGrow": 1,
              "justifyContent": "space-between",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flex": 1,
                  },
                  Object {
                    "paddingBottom": 12,
                    "paddingLeft": "6.25%",
                    "paddingRight": "6.25%",
                  },
                  Object {
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "marginBottom": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "rgb(0, 50, 120)",
                        "fontFamily": "Poppins",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "marginLeft": 2,
                        "marginTop": 10,
                      }
                    }
                  >
                    Send Wollo to
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignSelf": "stretch",
                          "borderColor": "rgb(127, 153, 188)",
                          "borderRadius": 22.5,
                          "borderWidth": 1,
                          "paddingLeft": 22,
                          "paddingRight": 22,
                        },
                        Object {
                          "borderRadius": 5,
                          "height": 97,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "rgb(0, 50, 120)",
                          "fontFamily": "Poppins",
                          "fontSize": 14,
                          "fontWeight": "bold",
                          "left": 22,
                          "opacity": 0.6,
                          "position": "absolute",
                          "top": 14,
                        }
                      }
                    >
                      Enter account key
                    </Text>
                    <TextInput
                      allowFontScaling={true}
                      autoCapitalize="sentences"
                      autoCorrect={true}
                      blurOnSubmit={true}
                      editable={true}
                      keyboardType="default"
                      multiline={true}
                      numberOfLines={3}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      placeholder=""
                      placeholderTextColor="rgb(127, 153, 188)"
                      returnKeyType="done"
                      style={
                        Array [
                          Object {
                            "color": "rgb(0, 50, 120)",
                            "flex": 1,
                            "fontFamily": "Poppins",
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "lineHeight": 21,
                            "textAlignVertical": "top",
                          },
                          Object {
                            "paddingTop": 12,
                            "width": "100%",
                          },
                        ]
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "position": "relative",
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../src/app/screens/send/images/wollo.png",
                      }
                    }
                    style={
                      Object {
                        "bottom": 21,
                        "height": 23,
                        "left": 11,
                        "position": "absolute",
                        "width": 23,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "marginBottom": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "rgb(0, 50, 120)",
                          "fontFamily": "Poppins",
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "marginBottom": 10,
                          "marginLeft": 2,
                          "marginTop": 10,
                        }
                      }
                    >
                      How much to send
                    </Text>
                    <View
                      style={
                        Array [
                          Object {
                            "alignSelf": "stretch",
                            "borderColor": "rgb(127, 153, 188)",
                            "borderRadius": 22.5,
                            "borderWidth": 1,
                            "paddingLeft": 22,
                            "paddingRight": 22,
                          },
                          Object {
                            "borderRadius": 25,
                            "height": 50,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "rgb(0, 50, 120)",
                            "fontFamily": "Poppins",
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "left": 22,
                            "opacity": 0.6,
                            "position": "absolute",
                            "top": 14,
                          }
                        }
                      >
                        Wollo
                      </Text>
                      <TextInput
                        allowFontScaling={true}
                        autoCapitalize="sentences"
                        autoCorrect={true}
                        blurOnSubmit={true}
                        editable={true}
                        keyboardType="numeric"
                        multiline={false}
                        numberOfLines={1}
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholder=""
                        placeholderTextColor="rgb(127, 153, 188)"
                        returnKeyType="done"
                        style={
                          Array [
                            Object {
                              "color": "rgb(0, 50, 120)",
                              "flex": 1,
                              "fontFamily": "Poppins",
                              "fontSize": 14,
                              "fontWeight": "bold",
                              "lineHeight": 21,
                              "textAlignVertical": "top",
                            },
                            Object {
                              "paddingTop": 10,
                              "width": "100%",
                            },
                            Object {
                              "paddingLeft": 42,
                            },
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "color": "rgb(127, 153, 188)",
                      "fontFamily": "Poppins",
                      "fontSize": 14,
                      "marginBottom": 5,
                    }
                  }
                >
                  Estimate
                   
                  £0, $0, €0, ¥0
                </Text>
                <View
                  style={
                    Object {
                      "marginBottom": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "rgb(0, 50, 120)",
                        "fontFamily": "Poppins",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "marginBottom": 10,
                        "marginLeft": 2,
                        "marginTop": 10,
                      }
                    }
                  >
                    Add a message?
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignSelf": "stretch",
                          "borderColor": "rgb(127, 153, 188)",
                          "borderRadius": 22.5,
                          "borderWidth": 1,
                          "paddingLeft": 22,
                          "paddingRight": 22,
                        },
                        Object {
                          "borderRadius": 5,
                          "height": 76,
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "rgb(0, 50, 120)",
                          "fontFamily": "Poppins",
                          "fontSize": 14,
                          "fontWeight": "bold",
                          "left": 22,
                          "opacity": 0.6,
                          "position": "absolute",
                          "top": 14,
                        }
                      }
                    >
                      Add a message to your Wollo transfer?
                    </Text>
                    <TextInput
                      allowFontScaling={true}
                      autoCapitalize="sentences"
                      autoCorrect={true}
                      blurOnSubmit={true}
                      editable={true}
                      keyboardType="default"
                      maxLength={28}
                      multiline={true}
                      numberOfLines={2}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      placeholder=""
                      placeholderTextColor="rgb(127, 153, 188)"
                      returnKeyType="done"
                      style={
                        Array [
                          Object {
                            "color": "rgb(0, 50, 120)",
                            "flex": 1,
                            "fontFamily": "Poppins",
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "lineHeight": 21,
                            "textAlignVertical": "top",
                          },
                          Object {
                            "paddingTop": 12,
                            "width": "100%",
                          },
                        ]
                      }
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "backgroundColor": "rgba(21, 74, 150, 0.5)",
                          "borderColor": "rgba(255, 255, 255, 0)",
                          "borderRadius": 22.5,
                          "borderWidth": 1,
                          "marginBottom": 10,
                          "maxWidth": 400,
                          "opacity": 1,
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "alignSelf": "stretch",
                              "color": "rgb(255, 255, 255)",
                              "fontFamily": "Poppins",
                              "fontSize": 14,
                              "fontWeight": "bold",
                              "height": 45,
                              "paddingLeft": 20,
                              "paddingRight": 20,
                              "paddingTop": 12,
                              "textAlign": "center",
                            },
                            undefined,
                            Array [
                              Object {
                                "color": "rgb(255, 255, 255)",
                              },
                              undefined,
                            ],
                            undefined,
                          ]
                        }
                      >
                        Next
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "borderColor": "rgb(0, 50, 120)",
                      "borderRadius": 22.5,
                      "borderWidth": 1,
                      "marginBottom": 10,
                      "maxWidth": 400,
                      "opacity": 1,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "alignSelf": "stretch",
                          "color": "rgb(255, 255, 255)",
                          "fontFamily": "Poppins",
                          "fontSize": 14,
                          "fontWeight": "bold",
                          "height": 45,
                          "paddingLeft": 20,
                          "paddingRight": 20,
                          "paddingTop": 12,
                          "textAlign": "center",
                        },
                        Object {
                          "color": "rgb(0, 50, 120)",
                        },
                        null,
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
